### PR TITLE
fixes #317 - add support for RequestID header

### LINKED ISSUE #317: Support reading the RequestID header on responses from Lightblue
The new header was added to Lightblue responses in https://github.com/lightblue-platform/lightblue-rest/pull/250. The client should make the value available to client applications.

### PR DESCRIPTION
**Pull request by dcrissman**
Makes headers available to client applications and also includes the RequestID in the error message.

### CODE REVIEW

File: core/src/main/java/com/redhat/lightblue/client/response/LightblueResponseException.java [hunk 1]
```diff
@@ -66,7 +68,12 @@ public boolean exists(String errorCode) {
     @Override
     public String getMessage() {
         if (lightblueResponse != null) {
-            return super.getMessage() + lightblueResponse.getText();
+            StringBuffer sb = new StringBuffer(super.getMessage());
+            if (!StringUtils.isEmpty(lightblueResponse.getRequestId())) {
+                sb.append(" RequestID:").append(lightblueResponse.getRequestId());
```
Comment (paterczm) [inline core/src/main/java/com/redhat/lightblue/client/response/LightblueResponseException.java]: If you use RequestID="<id>", splunk will extract it automatically.

File: core/src/test/java/com/redhat/lightblue/client/response/TestDefaultLightblueBulkDataResponse.java [hunk 2]
```diff
@@ -53,7 +53,7 @@ private DefaultLightblueBulkDataResponse simpleSetUp() throws Exception {
         bulkRequest.add(dfr);
         bulkRequest.add(dfr2);
 
-        return new DefaultLightblueBulkDataResponse(jsonResponse, bulkRequest);
+        return new DefaultLightblueBulkDataResponse(jsonResponse, null, bulkRequest);
```
Comment (paterczm) [inline core/src/test/java/com/redhat/lightblue/client/response/TestDefaultLightblueBulkDataResponse.java]: Maybe keep the DefaultLightblueBulkDataResponse(jsonResponse, bulkRequest) constructor for the cases where there are no headers?

File: http/src/main/java/com/redhat/lightblue/client/http/transport/Response.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package com.redhat.lightblue.client.http.transport;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Response {
```
Comment (paterczm) [inline http/src/main/java/com/redhat/lightblue/client/http/transport/Response.java]: I'd name it HttpResponse for consistency with other classes.

Comment (dcrissman) [inline http/src/main/java/com/redhat/lightblue/client/http/transport/Response.java]: done

File: core/src/main/java/com/redhat/lightblue/client/response/LightblueResponseException.java (hunk 1 above)
Comment (dcrissman) [inline core/src/main/java/com/redhat/lightblue/client/response/LightblueResponseException.java]: I didn't know that. Thanks for the suggestion. Done.

File: core/src/test/java/com/redhat/lightblue/client/response/TestDefaultLightblueBulkDataResponse.java (hunk 2 above)
Comment (dcrissman) [inline core/src/test/java/com/redhat/lightblue/client/response/TestDefaultLightblueBulkDataResponse.java]: Done.

**Pull request merged**